capi-tilt/README.md

```diff
@@ -0,0 +1,160 @@
+
+## Create tilt-settings.json
+
+Create a tilt-settings.json file and place it in your local copy of cluster-api. When using Kind, you’ll need a way to push your images to a registry to they can be pulled. Most users test with GCR, but you could also use something like Docker Hub. Here is an example:
+
+```
+{
+    "default_registry": "hub.docker.com/u/<user>",
+    "enable_providers": ["docker", "kubeadm-bootstrap", "kubeadm-control-plane"]
+}
+```
+
+## Running the tilt environment
+
+Run the tilt environment from your local copy of cluster api repo using the following command:
+
+``` 
+tilt up 
+``` 
+
+This will open the command-line HUD as well as a web browser interface. You can monitor Tilt’s status in either location. After a brief amount of time, you should have a running development environment, and you should now be able to create a cluster.
+
+## Create cluster
+
+In this folder you can find example .yaml files to run simple cluster for testing cluster api. First you need to create a **cluster** for your workflow cluster. CAPD is creating load-balancer container for the cluster, as it will be creating containers for all cluster resources.
+
+Create the cluster with:
+
+```
+kubectl apply -f cluster.yaml
+```
+Make sure cluster is provisioned.
+
+```
+kubectl get cluster
+```
+
+## Create control plane
+
+Next create 3 KubeadmControlPlanes with:
+
+```
+kubectl apply -f controlplane.yaml
+```
+Provisioning of the control planes will take time, cause they are created one-by-one. You can see the status of the deployment with:
+
+```
+kubectl get kcp
+```
+When provisioning is done. You should see:
+
+```
+NAME              READY   INITIALIZED   REPLICAS   READY REPLICAS   UPDATED REPLICAS   UNAVAILABLE REPLICAS
+my-controlplane           true          3          3                3 
+```
+
+## Copy working cluster Kubeconfig
+
+After the control plane node is up, you have to retrieve and save the working cluster Kubeconfig:
+
+```
+kubectl --namespace=default get secret/my-cluster-kubeconfig -o json | jq -r .data.value | base64 --decode > ./capi-kubeconfig
+```
+
+## Install and patch Calico 
+
+In this example we deploy Calico CNI solution. Deploy Calico using **capi-kubeconfig** copied in last step with:
+
+```
+kubectl --kubeconfig=capi-kubeconfig apply -f https://docs.projectcalico.org/v3.12/manifests/calico.yaml
+
+```
+After installation you need to patch Calico deployment to avoid issues with Docker provider. 
+
+```
+kubectl --kubeconfig=./capi-kubeconfig \
+  -n kube-system patch daemonset calico-node \
+  --type=strategic --patch='
+spec:
+  template:
+    spec:
+      containers:
+      - name: calico-node
+        env:
+        - name: FELIX_IGNORELOOSERPF
+          value: "true"
+'
+
+```
+
+## Create machinedeployment
+
+Finishing up the test cluster, create 3 node machinedeployment with:
+
+```
+kubectl apply -f machine-deployment.yaml
```

Comment:
change to
machine-deployments.yaml

Reply:
I changed the yaml name to machine-deployment.yaml